starfish/image/_filter/util.py

```diff
@@ -135,4 +135,4 @@ def preserve_float_range(array: Union[xr.DataArray, np.ndarray]) -> Union[xr.Dat
             array[array < 0] = 0
     if np.any(array > 1):
         array /= array.max()
-    return array
+    return array.astype(np.float32)
```

Comment:
why is this the right model?  shouldn't the input to this method already be a float32?

Reply:
Are there instances where this gets passed a non-float32 type?

Reply:
Yes, when e.g. a filter like Richardson Lucy casts to float 64. This way, we can just reduce precision at the end after we've already guaranteed we're in the right scale. 

I think this is the simplest place to add this guarantee of the right first size